src/Lib.P2P/PubSub/FloodRouter.cs

```diff
@@ -0,0 +1,316 @@
+                    }
+                }
+            };
+            try
+            {
+                var peers = SwarmService.KnownPeers.Where(p => p.ConnectedAddress != null);
+                await SendAsync(msg, peers, cancel).ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                _log.Warn("Leave topic failed.", e);
+            }
+        }
+
+        /// <inheritdoc />
+        public Task PublishAsync(PublishedMessage message, CancellationToken cancel)
+        {
+            if (_tracker.RecentlySeen(message.MessageId))
+            {
+                return Task.CompletedTask;
+            }
+
+            // Find a set of peers that are interested in the topic(s).
+            // Exclude author and sender
+            var peers = message.Topics
+               .SelectMany(topic => RemoteTopics.GetPeers(topic))
+               .Where(peer => peer != message.Sender)
+               .Where(peer => peer != message.Forwarder);
+
+            // Forward the message.
+            var forward = new PubSubMessage
+            {
+                PublishedMessages = new[] {message}
+            };
+
+            return SendAsync(forward, peers, cancel);
+        }
+
+        private Task SendAsync(PubSubMessage msg, IEnumerable<Peer> peers, CancellationToken cancel)
+        {
+            // Get binary representation
+            byte[] bin;
+            using (var ms = new MemoryStream())
+            {
+                Serializer.SerializeWithLengthPrefix(ms, msg, PrefixStyle.Base128);
+                bin = ms.ToArray();
+            }
+
+            return Task.WhenAll(peers.Select(p => SendAsync(bin, p, cancel)));
+        }
+
+        private async Task SendAsync(byte[] message, Peer peer, CancellationToken cancel)
+        {
+            try
+            {
+                await using (var stream = await SwarmService.DialAsync(peer, ToString(), cancel).ConfigureAwait(false))
+                {
+                    await stream.WriteAsync(message, 0, message.Length, cancel).ConfigureAwait(false);
+                    await stream.FlushAsync(cancel).ConfigureAwait(false);
+                }
+
+                _log.Debug($"sending message to {peer}");
+            }
+            catch (Exception e)
+            {
+                _log.Debug($"{peer} refused pubsub message.", e);
+            }
+        }
+
+#pragma warning disable VSTHRD100 // Avoid async void methods
+        /// <summary>
+        ///   Raised when a connection is established to a remote peer.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="connection"></param>
+        /// <remarks>
+        ///   Sends the hello message to the remote peer.  The message contains
+        ///   all topics that are of interest to the local peer.
+        /// </remarks>
+        private async void Swarm_ConnectionEstablished(object sender, PeerConnection connection)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this unused method parameter 'sender'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)